internal/provider/models/api_token.go

```diff
@@ -67,3 +86,48 @@ func (data *ApiTokenDataSource) ReadFromResponse(ctx context.Context, apiToken *
 	}
 	return diags
 }
+
+func (data *ApiTokenResource) ReadFromResponse(ctx context.Context, apiToken *iam.ApiToken) diag.Diagnostics {
+	var diags diag.Diagnostics
+	data.Id = types.StringValue(apiToken.Id)
+	data.Name = types.StringValue(apiToken.Name)
+	if apiToken.Description != "" {
+		data.Description = types.StringValue(apiToken.Description)
+	}
+	data.ShortToken = types.StringValue(apiToken.ShortToken)
+	data.Type = types.StringValue(string(apiToken.Type))
+	data.StartAt = types.StringValue(apiToken.StartAt.String())
+	if apiToken.EndAt != nil {
+		data.EndAt = types.StringValue(apiToken.EndAt.String())
+	} else {
+		data.EndAt = types.StringValue("")
+	}
+	data.CreatedAt = types.StringValue(apiToken.CreatedAt.String())
+	data.UpdatedAt = types.StringValue(apiToken.UpdatedAt.String())
+	data.CreatedBy, diags = SubjectProfileTypesObject(ctx, apiToken.CreatedBy)
+	if diags.HasError() {
+		return diags
+	}
+	data.UpdatedBy, diags = SubjectProfileTypesObject(ctx, apiToken.UpdatedBy)
+	if diags.HasError() {
+		return diags
+	}
+	if apiToken.ExpiryPeriodInDays != nil {
+		data.ExpiryPeriodInDays = types.Int64Value(int64(*apiToken.ExpiryPeriodInDays))
+	}
+	if apiToken.LastUsedAt != nil {
+		data.LastUsedAt = types.StringValue(apiToken.LastUsedAt.String())
+	} else {
+		data.LastUsedAt = types.StringValue("")
+	}
+	data.Roles, diags = utils.ObjectSet(ctx, apiToken.Roles, schemas.ApiTokenRoleAttributeTypes(), ApiTokenRoleTypesObject)
+	if diags.HasError() {
+		return diags
+	}
+	if apiToken.Token != nil {
+		data.Token = types.StringValue(*apiToken.Token)
+	} else {
+		data.Token = types.StringValue("")
```

Comment:
not sure how this will work since when you do anything other than create, the token value will not appear